### PR TITLE
Add support for A record based DNS SD

### PR DESCRIPTION
**Pull request by discordianfish**
If using A records, the user needs to specify "port" and set "type" to
"A".

### CODE REVIEW

**Comment by fabxc** [comment]
Let's make `SRV` the proper default in the config struct and be strict about the value, i.e. not everything other than `A` being `SRV`. This catches typos, too.
How about an error on non-zero port and `SRV` so people don't get the impression they could override the port from the record?

What about `AAAA` while we are at it?

LGTM in general. I'm curious where a use case for this came up? So far SRVs seemed to be quite ubiquitous.

**Comment by discordianfish** [comment]
@fabxc Does this look better? Also added AAAA lookup.

The use case in general is that in some cases you might not have a service in service discovery but just a list of hosts. Specifically I'll use that for monitoring a mesos cluster: Some basic OS level services like the node-exporter are not running on top of mesos but just as some service straight on the host. Therefor they are not available in the mesos state. Fortunately, mesos knows about it's master and slave nodes and with mesos-dns we have them available as A records.

File: config/config.go [hunk 1]
```diff
@@ -378,6 +380,9 @@ func (c *DNSSDConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	if len(c.Names) == 0 {
 		return fmt.Errorf("DNS-SD config must contain at least one SRV record name")
 	}
+	if strings.ToUpper(c.Type) != "SRV" && c.Port == 0 {
+		return fmt.Errorf("Port is required for all types except SRV records")
```
Comment (juliusv) [inline config/config.go]: Please start errors lower-cased. Also, some more context would be nice here. How about:

"a port is required in DNS-SD configs for all record types except SRV"

**Comment by juliusv** [comment]
:+1:

**Comment by fabxc** [comment]
:+1: after adding an error for a type that's not one of the three valid ones. Otherwise typos may go unnoticed and default yo `SRV`.

**Comment by discordianfish** [comment]
@fabxc It will print an error if the returned record is not a valid one. But I can't return an error in NewDNSDiscovery since it doesn't support returning an error, neither has ProvidersFromConfig in the targetmanager. So okay? ;)

**Comment by fabxc** [comment]
What about the config validation where the port error is returned?

**Comment by discordianfish** [comment]
@fabxc Good point. Like this?

**Comment by fabxc** [comment]
Perfect, thanks!

**Pull request merged**